src/MinimalDelegation.sol

```diff
@@ -53,6 +64,33 @@ contract MinimalDelegation is IERC7821, ERC1271, EIP712, ERC4337Account, Receive
         }
     }
 
+    /// @inheritdoc IERC7914
+    function approveNative(address spender, uint256 amount) external override returns (bool) {
+        _onlyThis();
+        MinimalDelegationStorageLib.get().allowance[spender] = amount;
+        emit ApproveNative(msg.sender, spender, amount);
+        return true;
+    }
+
+    /// @inheritdoc IERC7914
+    function transferFromNative(address from, address recipient, uint256 amount) public override returns (bool) {
+        if (from != address(this)) revert IncorrectSpender();
```

Comment:
nit: this is not IncorrectSpender() right? Since the `spender` is the msg.sender of this call

Reply:
i think `spender` is the `from` address

Reply:
`IncorrectSender()`?